common/port/matter_timers.c

```diff
@@ -238,6 +239,51 @@ void matter_timer_init(void)
     gtimer_start_periodical(&matter_rtc_timer, US_OVERFLOW_MAX, (void *)matter_timer_rtc_callback, (uint32_t) &matter_rtc_timer);
 }
 
+#if defined(CONFIG_ENABLE_AMEBA_SNTP) && (CONFIG_ENABLE_AMEBA_SNTP == 1)
+bool matter_sntp_rtc_is_sync(void)
+{
+    return matter_sntp_rtc_sync;
+}
+
+void matter_sntp_get_current_time(time_t *current_sec, time_t *current_usec)
+{
+    unsigned int update_tick = 0, retry = 0;
+    time_t update_sec = 0, update_usec = 0;
+    char* secKey  = "last-known-sntp-sec";
+    char* uSecKey = "last-known-sntp-usec";
+
+    sntp_get_lasttime(&update_sec, &update_usec, &update_tick);
+
+    if(update_tick) { //if sntp server is reachable, write to the dct and rtc
+        time_t tick_diff_sec, tick_diff_ms;
+        unsigned int current_tick = xTaskGetTickCount();
+
+        tick_diff_sec = (current_tick - update_tick) / configTICK_RATE_HZ;
+        tick_diff_ms = (current_tick - update_tick) % configTICK_RATE_HZ / portTICK_RATE_MS;
+        update_sec += tick_diff_sec;
+        update_usec += (tick_diff_ms * 1000);
+        *current_sec = update_sec + update_usec / 1000000;
+        *current_usec = update_usec % 1000000;
+        setPref_new(NULL,  secKey,  current_sec, sizeof(time_t));
```

Comment:
Why is it necessary to be stored into DCT?

Reply:
Removed the DCT usage in the latest commit. Instead it will read directly from the RTC which is already set by the InitClock_RealTime in the initialization.